### PR TITLE
add possibility to run sensitivity analysis whithout contengencies

### LINKED ISSUE #1528: Run sensitivity analysis without any contingencies
* **Do you want to request a *feature* or report a *bug*?**
Feature

* **What is the current behavior?**
To run a sensitivity analysis without any contingencies, the code should instanciate an `EmptyContingencyListProvider`.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**

* **What is the expected behavior?**
We can consider adding a new overload of the run / runAsync method to run a sensitivity analysis without passing this arguments. Thus, a user could write:
```
SensitivityAnalysisResult result = SensitivityAnalysis.run(network, factorsProvider, parameters);
```

This require to add to new methods in `SensitivityAnalysis` interface and two new methods in the `SensitivityAnalysis.Runner` class.

* **What is the motivation / use case for changing the behavior?**
Make the computation API more user-friendly

* **Please tell us about your environment:**
  - PowSyBl Version: 3.7.0
  - OS Version: ...

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, spectrum, etc)

(if a question doesn't apply, you can delete it)

### PR DESCRIPTION
**Pull request by jlabous**
Signed-off-by: Jérémy LABOUS <jlabous@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
fix #1528

### CODE REVIEW

**Review by murgeyseb — APPROVED** [review]

**Comment by murgeyseb** [comment]
Could you also create a PR on website documentation to adapt the tutorial accordingly (https://www.powsybl.org/pages/documentation/developer/tutorials/sensitivity-analysis.html)

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 1]
```diff
@@ -79,6 +79,32 @@ private Runner(SensitivityAnalysisProvider provider) {
             return runAsync(network, factorsProvider, contingenciesProvider, SensitivityAnalysisParameters.load());
         }
 
+        public CompletableFuture<SensitivityAnalysisResult> runAsync(Network network,
+                                                                     String workingStateId,
+                                                                     SensitivityAnalysisParameters parameters,
+                                                                     SensitivityFactorsProvider factorsProvider,
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:85]: Please keep the same order for the parameters. You also have to update the `runAsync` with all the parameters to assert that the contingenciesProvider is not null.

**Don't forget to add the `breaking change` label.**

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 2]
```diff
@@ -79,6 +79,32 @@ private Runner(SensitivityAnalysisProvider provider) {
...
+                                                                     ComputationManager computationManager) {
+            return runAsync(network, workingStateId, factorsProvider, new EmptyContingencyListProvider(), parameters, computationManager);
+        }
+
+        public CompletableFuture<SensitivityAnalysisResult> runAsync(Network network,
+                                                                     String workingStateId,
+                                                                     SensitivityAnalysisParameters parameters,
+                                                                     SensitivityFactorsProvider factorsProvider) {
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:93]: Same remark about the parameter order

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 3]
```diff
@@ -79,6 +79,32 @@ private Runner(SensitivityAnalysisProvider provider) {
...
+            return runAsync(network, workingStateId, parameters, factorsProvider, DefaultComputationManagerConfig.load().createLongTimeExecutionComputationManager());
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:94]: I think, if the contingency list is empty we should use the short-time execution computation manager.

@murgeyseb Do you agree?

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 4]
```diff
@@ -79,6 +79,32 @@ private Runner(SensitivityAnalysisProvider provider) {
...
+        }
+
+        public CompletableFuture<SensitivityAnalysisResult> runAsync(Network network,
+                                                                     SensitivityAnalysisParameters parameters,
+                                                                     SensitivityFactorsProvider factorsProvider) {
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:99]: Same remark about the parameter order

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 5]
```diff
@@ -109,6 +135,32 @@ public SensitivityAnalysisResult run(Network network,
...
+                                             SensitivityAnalysisParameters parameters,
+                                             SensitivityFactorsProvider factorsProvider) {
+            return runAsync(network, workingStateId, factorsProvider, new EmptyContingencyListProvider(), parameters).join();
+        }
+
+        public SensitivityAnalysisResult run(Network network,
+                                             SensitivityAnalysisParameters parameters,
+                                             SensitivityFactorsProvider factorsProvider) {
+            return runAsync(network, factorsProvider, new EmptyContingencyListProvider(), parameters).join();
+        }
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:157]: Same remark about the parameter order

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 6]
```diff
@@ -163,4 +215,22 @@ public static SensitivityAnalysisResult run(Network network,
...
+                                                SensitivityAnalysisParameters parameters,
+                                                SensitivityFactorsProvider factorsProvider) {
+        return find().run(network, workingStateId, parameters, factorsProvider);
+    }
+
+    public static SensitivityAnalysisResult run(Network network,
+                                                SensitivityAnalysisParameters parameters,
+                                                SensitivityFactorsProvider factorsProvider) {
+        return find().run(network, parameters, factorsProvider);
+    }
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:230]: Same remark about the parameter order. It seems that all the runAsync methods are missing.

@murgeyseb Do we add them all? Only some of them? I think we should be symmetrical and provide only part of them (the most useful) but in sync/async modes

**Review by mathbagu — CHANGES_REQUESTED** [review]

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java (hunk 3 above)
Comment (murgeyseb) [reply sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:94]: Yes, why not.

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java (hunk 6 above)
Comment (murgeyseb) [reply sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:230]: It should definitely be symmetrical between async and sync.

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 7]
```diff
@@ -109,6 +135,32 @@ public SensitivityAnalysisResult run(Network network,
...
+                                             String workingStateId,
+                                             SensitivityFactorsProvider factorsProvider,
+                                             SensitivityAnalysisParameters parameters) {
+            return runAsync(network, workingStateId, factorsProvider, new EmptyContingencyListProvider(), parameters).join();
+        }
+
+        public SensitivityAnalysisResult run(Network network,
+                                             SensitivityFactorsProvider factorsProvider,
+                                             SensitivityAnalysisParameters parameters) {
+            return runAsync(network, factorsProvider, new EmptyContingencyListProvider(), parameters).join();
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:156]: These methods should call the corresponding `runAsync` method without adding new parameters

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java [hunk 8]
```diff
@@ -163,4 +224,30 @@ public static SensitivityAnalysisResult run(Network network,
...
+    public static SensitivityAnalysisResult run(Network network,
+                                                SensitivityFactorsProvider factorsProvider,
+                                                SensitivityAnalysisParameters parameters) {
+        return find().run(network, factorsProvider, parameters);
+    }
+
+    public static SensitivityAnalysisResult run(Network network,
+                                                SensitivityFactorsProvider factorsProvider) {
+        return find().run(network, factorsProvider);
+    }
```
Comment (mathbagu) [inline sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:252]: As we ask in the review, can you add all missing runAsync method please?

**Review by mathbagu — CHANGES_REQUESTED** [review]

File: sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java (hunk 8 above)
Comment (jlabous) [reply sensitivity-analysis-api/src/main/java/com/powsybl/sensitivity/SensitivityAnalysis.java:252]: I think we get all methods runAsync
We have in the Runner 8 methods runAsync with 8 corresponding methods run
We also hase 8 corresponding methods run in sensitivity Analysis

**Review by mathbagu — APPROVED** [review]

**Pull request merged**